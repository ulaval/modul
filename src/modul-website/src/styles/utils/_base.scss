html,
body,
.modul-body {
    background: $modul-color--body;
    height: 100%;
}
