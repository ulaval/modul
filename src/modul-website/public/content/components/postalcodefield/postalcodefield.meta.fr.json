{
    "name": "Champ de code postal",
    "url": "champs-de-code-postal",
    "category": "forms",
    "type": "components",
    "visible": true,
    "preview": "components/postalcodefield/postalcodefield.preview.fr.md",
    "components": [
        "m-postalcodefield"
    ]
}
