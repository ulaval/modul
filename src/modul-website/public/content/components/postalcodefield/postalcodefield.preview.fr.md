<m-postalcodefield label="code postal"></m-postalcodefield>
