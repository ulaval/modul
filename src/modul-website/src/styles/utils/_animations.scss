.modul-u--light-template-transition {
    &-enter-active,
    &-leave-active {
        @media (min-width: #{$m-mq--min--s}) {
            transition: transform $m-transition-duration--l ease, opacity $m-transition-duration--l ease;
        }
    }

    &-enter,
    &-leave-to {
        @media (min-width: #{$m-mq--min--s}) {
            transform: translate3d(0, 44px, 0);
            opacity: 0.6;
        }
    }
}
