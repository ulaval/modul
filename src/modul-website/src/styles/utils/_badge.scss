.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}

.modul-logo__new {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--hint;
    padding: 2px 3px;
}
