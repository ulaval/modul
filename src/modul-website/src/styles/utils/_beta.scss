.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}
