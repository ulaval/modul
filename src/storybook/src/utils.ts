export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
