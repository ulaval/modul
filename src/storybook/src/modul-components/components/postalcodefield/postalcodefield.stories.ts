import { POSTALCODEFIELD_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MPostalCodeCountry } from '@ulaval/modul-components/dist/components/postalcodefield/postalcodefield';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${POSTALCODEFIELD_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    data: () => ({
        model: 'asdfa2341234'
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model"></${POSTALCODEFIELD_NAME}>`
});

defaultStory.story = {
    name: 'Default'
};

export const withValue = () => ({
    data: () => ({
        model: 'G1V0A6'
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model"></${POSTALCODEFIELD_NAME}>`
});

export const other = () => ({
    data: () => ({
        model: 'G1V0A6',
        postalCodeFormat: MPostalCodeCountry.Other
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model" :postal-code-format="postalCodeFormat"></${POSTALCODEFIELD_NAME}>`
});

export const UnitedStates = () => ({
    data: () => ({
        model: '10001',
        postalCodeFormat: MPostalCodeCountry.US
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model" :postal-code-format="postalCodeFormat"></${POSTALCODEFIELD_NAME}>`
});

export const France = () => ({
    data: () => ({
        model: '75012',
        postalCodeFormat: MPostalCodeCountry.FR
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model" :postal-code-format="postalCodeFormat"></${POSTALCODEFIELD_NAME}>`
});

export const withNotPostalCodeDefaultValue = () => ({
    data: () => ({
        model: 'ad23dgvb3ssdffdd'
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model"></${POSTALCODEFIELD_NAME}>`
});

export const withLabelAndError = () => ({
    data: () => ({
        model: '123456'
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model" label="Postal code" :error="true" error-message="Invalid postal code with a long long long long long message error"></${POSTALCODEFIELD_NAME}>`
});

export const withLabelUpAndFullWidth = () => ({
    data: () => ({
        model: ''
    }),
    template: `<${POSTALCODEFIELD_NAME} v-model="model" label="Postal code" :label-up="true" max-width="100%"></${POSTALCODEFIELD_NAME}>`
});
