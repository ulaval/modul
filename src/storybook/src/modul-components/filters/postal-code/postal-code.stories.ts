import { text } from '@storybook/addon-knobs';
import { MPostalCodeCountry } from '@ulaval/modul-components/dist/components/postalcodefield/postalcodefield';
import { POSTAL_CODE_NAME } from '@ulaval/modul-components/dist/filters/filter-names';
import PostalCodeFilterPlugin from '@ulaval/modul-components/dist/filters/postal-code/postal-code';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

Vue.use(PostalCodeFilterPlugin);

let template: string = `<div>{{ text | f-m-postal-code(isoCountry) }}</div>`;

export default {
    title: `${modulComponentsHierarchyRootSeparator}${POSTAL_CODE_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    props: {
        text: {
            default: text('Text', '9999999')
        }
    },
    template: template
});

defaultStory.story = {
    name: 'Default'
};

export const ca = () => ({
    props: {
        text: {
            default: text('Text', 'G2B0B9')
        },
        isoCountry: {
            default: MPostalCodeCountry.CA
        }
    },
    template: template
});

export const us5Characters = () => ({
    props: {
        text: {
            default: text('Text', '99999')
        },
        isoCountry: {
            default: MPostalCodeCountry.US
        }
    },
    template
});

export const us9Characters = () => ({
    props: {
        text: {
            default: text('Text', '999999999')
        },
        isoCountry: {
            default: MPostalCodeCountry.US
        }
    },
    template
});

export const fr5Characters = () => ({
    props: {
        text: {
            default: text('Text', '99999')
        },
        isoCountry: {
            default: MPostalCodeCountry.FR
        }
    },
    template
});
