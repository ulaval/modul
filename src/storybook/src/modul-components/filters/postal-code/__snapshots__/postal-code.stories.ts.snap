// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|f-m-postal-code Ca 1`] = `
<div>
  G2B 0B9
</div>
`;

exports[`Storyshots modul-components|f-m-postal-code Default 1`] = `
<div>
  9999999
</div>
`;

exports[`Storyshots modul-components|f-m-postal-code Fr 5 Characters 1`] = `
<div>
  99999
</div>
`;

exports[`Storyshots modul-components|f-m-postal-code Us 5 Characters 1`] = `
<div>
  99999
</div>
`;

exports[`Storyshots modul-components|f-m-postal-code Us 9 Characters 1`] = `
<div>
  99999-9999
</div>
`;
